node-src/git/getCommitAndBranch.ts

```diff
@@ -56,7 +56,7 @@ export default async function getCommitAndBranch(
   const isGitHubAction = GITHUB_ACTIONS === 'true';
   const isGitHubPrBuild = GITHUB_EVENT_NAME === 'pull_request';
 
-  if (!(await hasPreviousCommit())) {
+  if (!options.allowSingleCommitRepositories && !(await hasPreviousCommit())) {
```

Comment:
Perhaps we should just only run the check in CI?

We could even query for builds _not_ matching the commit and throw if we find something (which indicates other commits are known to exist). That's not completely foolproof though (they might've amended) and probably isn't worth it.

Reply:
Either way I think we should still print a warning message.